RIOT-OS/RIOT
# cpu/kinetis: model kconfig

### Contribution description
This models the Kconfig modules for all boards based on kinetis. 72e7f8d30de877970694c14f9b6a5ab313bb8111 limits the CI to test only this on `mulle` for now. Some changes on drivers are also included here (mostly adding hardware features to indicate presence).

The list of enabled boards is:
- frdm-k22f
- frdm-k64f
- frdm-kl43z
- frdm-kw41z
- mulle
- openlabs-kw41z-mini
- openlabs-kw41z-mini-256kib
- pba-d-01-kw2x
- phynode-kw41z
- teensy31
- usb-kw41z

### Testing procedure
- Green CI
- Check the modelling and compare `make info-modules` list with and without `TEST_KCONFIG=1`, they should match.
- Double-check that all boards based on kinetis are here

### Issues/PRs references
Part of #16875
~~Depends on #17307~~
